src/java.base/share/classes/java/nio/channels/spi/AbstractSelectableChannel.java

```diff
@@ -230,7 +230,10 @@ public final SelectionKey register(Selector sel, int ops, Object att)
                 SelectionKey k = findKey(sel);
                 if (k != null) {
                     k.attach(att);
-                    k.interestOps(ops);
+                    try {
+                        k.interestOps(ops);
+                    } catch (CancelledKeyException ignored) {
+                    }
```

Comment:
The register method is specified to throw CKE "If this channel is currently registered with the given selector but the corresponding key has already been cancelled". So we shouldn't be changing anything in this class.

Reply:
Reverted in 98bcd22e8ba60c7499fab332f943616bb88b03f5.